### ISSUE TITLE
NPM crashes when deleting a few hosts and ssl certs, then resets to default and it is empty without any records

### ISSUE THREAD

**Issue by comfuzio** [NginxProxyManager/nginx-proxy-manager#3610]
Hi, I am running NPM for more than 1 year now in a proxmox vm from template of ubuntu 22.04.3 server (full updated).
I have installed docker via the official docker instructions and then portainer for gui.
My docker compose file is this:
`version: '3.3'

services:
  app:
    image: 'jc21/nginx-proxy-manager:latest'
    restart: unless-stopped
    ports:
      - '80:80'
      - '81:81'
      - '443:443'
    volumes:
      - ./data:/data
      - ./letsencrypt:/etc/letsencrypt

  goaccess:
    image: 'xavierh/goaccess-for-nginxproxymanager:latest'
    container_name: goaccess
    restart: always
    ports:
      - '7880:7880'
    environment:
      - PUID=0 
      - PGID=0
      - TZ=Europe/Athens
      - SKIP_ARCHIVED_LOGS=False # optional
      - DEBUG=False # optional
      - BASIC_AUTH=False # optional
      - BASIC_AUTH_USERNAME=user # optional
      - BASIC_AUTH_PASSWORD=pass # optional
      - EXCLUDE_IPS=127.0.0.1 # optional - comma delimited
      - LOG_TYPE=NPM # optional - more information below
    volumes:
      - ./data/logs:/opt/log
`
to clarify the problem which I will describe happens also before I have added the goaccess to my npm compose.
So I have a few domain names and some subdomains etc and it works perfect and thank you for this amazing project!
The problem happens now that I am moving my websites to another VM running a webhosting software than keeping them in docker compose. 
I have migrated all my sites to the new host, change A records to match the new IP and all runs good!
So I go to NPM to delete those sites, if I delete 1-2 domains and their certs, I don't see any problem but when I try to delete more, one at a time of course, NPM is not responding.
So I stop the stack and restart it, when it starts, it is in factory default settings! I have to login with the default info and it doesn't contain any record at all! 
So I roll back to the backup I am always taking before I do such actions.

**Checklist**
- Have you pulled and found the error with `jc21/nginx-proxy-manager:latest` docker image?
  - Yes
- Are you sure you're not using someone else's docker image?
  - Yes
- Have you searched for similar issues (both open and closed)?
  - Yes

**Describe the bug**

NPM crashes when deleting hosts and certs and when re-started it is in factory default state.

**Nginx Proxy Manager Version**

v2.11.1

**To Reproduce**
1. Go to 'Hosts>proxy hosts'
2. Click on '3 dots on a host and delete'
3. Go to 'SSL Certificates'
4. Delete the matching record.
5. Repeat that 5 times or so.
6. NPM is not responding.
7. Restart service and NPM is in factory defaults.

**Expected behavior**

Delete the hosts and certifications and not crash or lose all remaining records after restart

**Operating System**

Ubuntu 22.04.3 server full updated on proxmox 8.1 host

**notes**
I had an ARM64 vm debian 12 in hetzner with same installation of docker and portainer and npm with this yaml and it did the same there as well.
It did the same even before i discovered the goaccess

**Comment by comfuzio** [comment]
most of them were in the style of domain.com and www.domain.com but some where subdomains such as subdomain.domain.com.
Indeed some of the certs if I go to the certs tab are red and outdated, this is also odd because NPM never updates them there, the certs themselves if you visit the sites, they were autorenew but the certs tab didn't seem to get the update

**Comment by comfuzio** [comment]
For now I just want to remove all those domains that I have transferred to the other vm and just keep those that remain in the same portainer instance (they still are plenty).
My workaround for now seems to be to make a record of what I want to keep and the internal ip/port and redo them all since I don't want to create another interruption to the transferred domains to change the A record back, do SSL renew, change A record again.

**Comment by comfuzio** [comment]
Is there any update on this issue? if not, it is safe to close it. As an issue, for me it still remains unfortunatelly.

**Comment by dominicm00** [comment]
Can confirm this is still happening 👍🏼

**Comment by sourcecodemage** [comment]
Is there a manual fix for this? 

`2026-04-26 03:14:25.282894+00:00nginx: [emerg] cannot load certificate "/etc/letsencrypt/live/npm-13/fullchain.pem": BIO_new_file() failed (SSL: error:80000002:system library::No such file or directory:calling fopen(/etc/letsencrypt/live/npm-13/fullchain.pem, r) error:10000080:BIO routines::no such file)
2026-04-26 03:14:26.293803+00:00[1;34m❯ [1;36mStarting nginx ...[0m
2026-04-26 03:14:26.300893+00:00nginx: [warn] the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /etc/nginx/nginx.conf:4
`

This just loops and I can't get to the web interface. I'm running NPM as  TrueNAS application.

**Comment by dominicm00** [comment]
> Is there a manual fix for this?

I believe I manually edited entries in the config DB to not point to this certificate; don't remember the exact steps I took 😢

**Comment by BladedNarwhal** [comment]
In my case, I had deleted my certificate for a domain I was no longer using, but like a dummy didn't also delete the corresponding Proxy Host. I attempted remaking the letsencrypt live folder for the cert I deleted and using touch to make dummy .pem files in attempt to bypass the crash, however this did not work. Thankfully I don't have all too many proxy hosts and was able to manually open each .conf to find the one corresponding to the certificate I deleted and delete the .conf file. This then fixed the issue for me. Hope this helps somebody else!

TLDR;
I deleted certificate but forgot about proxy host. Manually deleted proxy host .conf file for corresponding certificate and issue was resolved.
